[Explore] Introduce facet filter

Maosaic opened the pull request:
### Description

<!-- Describe what this change achieves-->

What are facets?

To help users quickly narrow down their search data, facets will provide access filtering commonly understood fields. Currently, we are introducing 3 default facet field(`serviceName`/`span.attributes.http@status_code`/`status.code`).

This PR introduces the UI components of facets filter.

### Issues Resolved

<!-- List any issues this PR will resolve. Prefix the issue with the keyword closes, fixes, fix -->
<!-- Example: closes #1234 or fixes <Issue_URL> -->

## Screenshot

<!-- Attach any relevant screenshots. Any change to the UI requires an attached screenshot in the PR Description -->


https://github.com/user-attachments/assets/73bc6a9a-8e5f-4340-90bf-d21b96245f8b



## Testing the changes

<!--
  Please provide detailed steps for validating your changes. This could involve specific commands to run,
  pages to visit, scenarios to try or any other information that would help reviewers verify
  the functionality of your change
-->

## Changelog
<!--
Add a short but concise sentence about the impact of this pull request. Prefix an entry with the type of change they correspond to: breaking, chore, deprecate, doc, feat, fix, infra, refactor, test.
- fix: Update the graph
- feat: Add a new feature

If this change does not need to added to the changelog, just add a single `skip` line e.g.
- skip

Descriptions following the prefixes must be 100 characters long or less
-->

- feat: Introduce facet filter

### Check List

- [ ] All tests pass
  - [ ] `yarn test:jest`
  - [ ] `yarn test:jest_integration`
- [ ] New functionality includes testing.
- [ ] New functionality has been documented.
- [ ] Update [CHANGELOG.md](./../CHANGELOG.md)
- [ ] Commits are signed per the DCO using --signoff

TackAdam commented:
1. Should we be hiding this facet field when on the logs flavor? We only want this displayed while on traces
<img width="1522" height="649" alt="Screenshot 2025-08-07 at 11 39 39 AM" src="https://github.com/user-attachments/assets/40dc0d21-6339-459b-9e76-61bd54583871" />
2. For the status.code section is there a way we can alias the values to be as follows

```
0 = Unset
1 = Ok
2 = Error
```
https://github.com/open-telemetry/opentelemetry-specification/blob/v1.46.0/specification/trace/api.md#set-status
<img width="220" height="112" alt="Screenshot 2025-08-07 at 11 41 01 AM" src="https://github.com/user-attachments/assets/11a38b8d-92e7-454d-892d-095bb62d3187" />
3. I don't see the facet for "attributes.http.status_code" showing up by default
<img width="1251" height="684" alt="Screenshot 2025-08-07 at 11 45 51 AM" src="https://github.com/user-attachments/assets/4fc4a806-036c-4cfc-8f87-dcc0cc705416" />
4. I think if we are able to alias the name it would make sense for status.code to be "OTel.status" and for attributes.http.status_code to just be "http.status" and serviceName should just be "Services (x)"
5. Can we adjust the order so the two error fields show up first and then the "Services (x)"

TackAdam commented:
Offline discussion. LGTM just pending switch from 'span.attributes.http@status_code' to 'attributes.http.status_code'

Maosaic commented:
> 1. Should we be hiding this facet field when on the logs flavor? We only want this displayed while on traces
> 
> 2. For the status.code section is there a way we can alias the values to be as follows
> ```
> 0 = Unset
> 1 = Ok
> 2 = Error
> ```
> 
> 3. I don't see the facet for "attributes.http.status_code" showing up by default 
>
> 4. I think if we are able to alias the name it would make sense for status.code to be "OTel.status" and for attributes.http.status_code to just be "http.status" and serviceName should just be "Services (x)" 
>
>5. Can we adjust the order so the two error fields show up first and then the "Services (x)"

1#&4# will need more discussion, 2# can be achieved by index pattern field mapping, updated the mapping for 3#, 5# the default order is sort by value hit count, also updated the UI to display count.
